packages/core/src/electron-browser/menu/electron-menu-contribution.ts

```diff
@@ -77,11 +77,21 @@ export class ElectronMenuContribution implements FrontendApplicationContribution
             }
         }
 
+        const currentWindow = electron.remote.getCurrentWindow();
         const createdMenuBar = this.factory.createMenuBar();
+
         if (isOSX) {
             electron.remote.Menu.setApplicationMenu(createdMenuBar);
+            currentWindow.on('focus', () =>
```

Comment:
I think the listener should be removed on window `'close'`. Thoughts?

Reply:
I would think that upon closing a window, callbacks attached to it would be garbage collected along with the destroyed window, but for sanity I could do that.

Code after modification:
```ts
        if (isOSX) {
            electron.remote.Menu.setApplicationMenu(createdMenuBar);
            const focusCallback = () => {
                // OSX: Recreate the menus when changing windows.
                // OSX only has one menu bar for all windows, so we need to swap
                // between them as the user switch windows.
                electron.remote.Menu.setApplicationMenu(this.factory.createMenuBar());
            };
            currentWindow.on('focus', focusCallback);
            currentWindow.once('closed', () => {
                currentWindow.removeListener('closed', focusCallback);
            });
        }
```

Looks kinda uglier.

Reply:
Are you sure we should go with the listener cleanup, that should be destroyed with the window anyway?

Reply:
OK.